ekf/tests/test_runner/result_check.c

```diff
@@ -0,0 +1,307 @@
+	err |= parser_fieldGetFloat(h, "velocity_z", &ekfTestsResult_common.expectedState->data[VZ]);
+
+	err |= parser_fieldGetFloat(h, "accel_bias_x", &ekfTestsResult_common.expectedState->data[BAX]);
+	err |= parser_fieldGetFloat(h, "accel_bias_y", &ekfTestsResult_common.expectedState->data[BAY]);
+	err |= parser_fieldGetFloat(h, "accel_bias_z", &ekfTestsResult_common.expectedState->data[BAZ]);
+
+	err |= parser_fieldGetFloat(h, "position_x", &ekfTestsResult_common.expectedState->data[RX]);
+	err |= parser_fieldGetFloat(h, "position_y", &ekfTestsResult_common.expectedState->data[RY]);
+	err |= parser_fieldGetFloat(h, "position_z", &ekfTestsResult_common.expectedState->data[RZ]);
+
+	err |= parser_fieldGetFloat(h, "attitude_eps", &eps);
+	ekfTestsResult_common.eps->data[QA] = eps;
+	ekfTestsResult_common.eps->data[QB] = eps;
+	ekfTestsResult_common.eps->data[QC] = eps;
+	ekfTestsResult_common.eps->data[QD] = eps;
+
+	err |= parser_fieldGetFloat(h, "gyro_bias_eps", &eps);
+	ekfTestsResult_common.eps->data[BWX] = eps;
+	ekfTestsResult_common.eps->data[BWY] = eps;
+	ekfTestsResult_common.eps->data[BWZ] = eps;
+
+	err |= parser_fieldGetFloat(h, "velocity_eps", &eps);
+	ekfTestsResult_common.eps->data[VX] = eps;
+	ekfTestsResult_common.eps->data[VY] = eps;
+	ekfTestsResult_common.eps->data[VZ] = eps;
+
+	err |= parser_fieldGetFloat(h, "accel_bias_eps", &eps);
+	ekfTestsResult_common.eps->data[BAX] = eps;
+	ekfTestsResult_common.eps->data[BAY] = eps;
+	ekfTestsResult_common.eps->data[BAZ] = eps;
+
+	err |= parser_fieldGetFloat(h, "position_eps", &eps);
+	ekfTestsResult_common.eps->data[RX] = eps;
+	ekfTestsResult_common.eps->data[RY] = eps;
+	ekfTestsResult_common.eps->data[RZ] = eps;
+
+	return err;
+}
+
+
+static int ekftests_expectedResultGet(const char *expectedResultFile, matrix_t *expectedState, matrix_t *eps)
+{
+	parser_t *parser;
+	int res;
+
+	ekfTestsResult_common.expectedState = expectedState;
+	ekfTestsResult_common.eps = eps;
+
+	parser = parser_alloc(1, STATE_LENGTH + EPS_FIELDS_NB);
+	if (parser == NULL) {
+		return -1;
+	}
+
+	if (parser_headerAdd(parser, "EXP_EKF_STATUS", ekftests_expectedResultConverter) != 0) {
+		parser_free(parser);
+		return -1;
+	}
+
+	res = parser_execute(parser, expectedResultFile, PARSER_IGN_UNKNOWN_HEADERS);
+	parser_free(parser);
+
+	return res;
+}
+
+
+static int ekftests_finalStateGet(const char *logFile, matrix_t *finalState)
+{
+	/* Timestamp is needed only to pass valid pointer to `ekflog_stateRead`. Currently we are not checking timestamps */
+	time_t timestamp;
+
+	if (ekflog_readerInit(logFile) != 0) {
+		return -1;
+	}
+
+	errno = 0;
+
+	while (ekflog_stateRead(finalState, &timestamp) == 0)
+		;
+	if (ekflog_readerDone() != 0) {
+		return -1;
+	}
+
+	/* `ekflog_stateRead` sets errno in case of error */
+	return errno;
+}
+
+
+static int ekftests_statesAlloc(matrix_t *state1, matrix_t *state2, matrix_t *state3, matrix_t *state4)
```

Comment:
It seems that four copies of state matrix are arbitrary. The cause for four copies is hidden in the upper layer. I think that each state purpose can be highlighted in its name:
```suggestion
static int ekftests_statesAlloc(matrix_t *expected, matrix_t *final, matrix_t *epsilon, matrix_t *diff)
```